### ISSUE TITLE
Overlapping stack structures have really weird failure mode

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#2063]
Windows 10 Personal 2491-dev

Consider this program:
```C
#include <stdint.h>
#include <stdio.h>

struct A {
        uint64_t one;
        uint64_t two;
};

void quux(struct A *x, struct A *y) {
        printf("%d\n", x->one + y->two);
        printf("%d\n", x->two + y->one);
}

int main(void) {
        struct A p = { 0x41, 0x42 };
        struct A q = { 0x51, 0x52 };

        quux(&p, &q);
}
// gcc -o struct-overlap struct-overlap.c
```

Now let's imagine RE'ing it binja, only we get the structure definition wrong:
```
struct A __packed
{
    uint64_t one;
    uint64_t two;
    uint8_t wrong[0x10];
};
```

```
int32_t main(int32_t arg1, char** arg2, char** arg3)

0000119c  uint64_t var_18 = 0x41
000011ac  uint64_t var_28 = 0x51
000011ca  quux(&var_18, &var_28)
000011d5  return 0
```

We set the function type on quux to be correct: `void quux(struct A* arg1, struct A* arg2)`, but notably the stack based args are not correct: they are both uint64_t's. If we set `var_28` to the correct type (but with the wrong definition) things get really weird:
```
int32_t main(int32_t arg1, char** arg2, char** arg3)

0000119c  struct A var_28
0000119c  var_28.wrong[0].q = 0x41
000011a4  var_28.wrong[8].q = 0x42
000011ac  var_28.one = 0x51
000011b4  var_28.two = 0x52
000011ca  var_18
000011ca  quux(&var_18, &var_28)
000011d5  return 0
```

Notably, `var_18` has no type or declaration, it's just kinda... hanging out? And in fact, if we try to change `var_18`'s type, we get a change type dialog for the main function, not the variable itself. This is obviously a contrived case, but having incorrect structure definitions/sizes is reasonably common when RE'ing, so some so a more graceful failure in this case would be useful.

[struct-overlap.elf.txt](https://github.com/Vector35/binaryninja-api/files/5434192/struct-overlap.elf.txt)

**Comment by yrp604** [comment]
With some more thought, this is probably a dupe/slight variant of #1649.

**Comment by plafosse** [comment]
This was fixed along with #1649
